src/lib/index.js

```diff
@@ -2,15 +2,43 @@
 
 var d3 = require('@plotly/d3');
 var utcFormat = require('d3-time-format').utcFormat;
+var d3Format = require('d3-format').format;
 var isNumeric = require('fast-isnumeric');
 
 var numConstants = require('../constants/numerical');
 var MAX_SAFE = numConstants.FP_SAFE;
 var MIN_SAFE = -MAX_SAFE;
 var BADNUM = numConstants.BADNUM;
 
-var lib = module.exports = {
-    _numberFormat: d3.format // simply to test d3.format before switching to d3-format
+var lib = module.exports = {};
+
+lib.adjustFormat = function adjustFormat(formatStr) {
+    if(
+        !formatStr ||
+        /^[0123456789].[0123456789]f/.test(formatStr) ||
+        /.[0123456789]%/.test(formatStr)
+    ) return formatStr;
+
+    if(formatStr === '0.f') return '~f';
+    if(/^[0123456789]%/.test(formatStr)) return '~%';
+    if(/^[0123456789]s/.test(formatStr)) return '~s';
```

Comment:
```suggestion
        /^\d[.]\df/.test(formatStr) ||
        /[.]\d%/.test(formatStr)
    ) return formatStr;

    if(formatStr === '0.f') return '~f';
    if(/^\d%/.test(formatStr)) return '~%';
    if(/^\ds/.test(formatStr)) return '~s';
```

ie `[0123456789]` === `\d` and you don't mean `.` (any character), you mean `[.]` (just a period character)